tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -79,6 +79,67 @@ def test_slurm_gpu(region, pcluster_config_reader, clusters_factory):
     assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
 
 
+@pytest.mark.regions(["us-east-2"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["slurm"])
+@pytest.mark.skip_oss(["centos6"])
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+@pytest.mark.slurm_mpi
+def test_slurm_mpi(region, pcluster_config_reader, clusters_factory, test_datadir):
+    """
+    Test Slurm with mpirun.
+
+    mpirun job is known to leave stray processes after job termination if slurm process tracking is not setup correctly,
+    i.e. using ProctrackType=proctrack/pgid
+    Test mpirun script to make sure no stray processes after the job is terminated due to timeout
+    Test should run on all OSs except for centos6, where mpirun is not available
+    """
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    _test_mpi_job_termination(remote_command_executor, test_datadir)
+
+    assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
+
+
+def _test_mpi_job_termination(remote_command_executor, test_datadir):
+    """Test canceling mpirun job will not leave stray processes."""
+    logging.info("Testing no stray process left behind after mpirun job is terminated")
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    # Submit mpi_job, leaving 1 vcpu on each node idle so that the process check job can run while mpi_job is running
+    # Note: using 5 compute nodes for mpi_job so that the job takes more time to complete than the jobs's time limit
+    # (1 min, which is the lowest time limit possible)
+    # Want mpi_job to timeout to test that the termination signals are sent to the child MPI processes
+    result = slurm_commands.submit_script(str(test_datadir / "mpi_job.sh"))
+    job_id = slurm_commands.assert_job_submitted(result.stdout)
+
+    # Check that mpi processes are started
+    _check_mpi_process(remote_command_executor, slurm_commands, test_datadir, after_completion=False)
+    slurm_commands.wait_job_completed(job_id)
+
+    # Make sure mpirun job is stopped due to timeout
+    _assert_job_state(remote_command_executor, job_id, job_state="TIMEOUT")
+
+    # Check that mpi processes are terminated
+    _check_mpi_process(remote_command_executor, slurm_commands, test_datadir, after_completion=True)
+
+
+def _check_mpi_process(remote_command_executor, slurm_commands, test_datadir, after_completion):
+    """Submit script and check for MPI processes."""
+    # Clean up old datafiles
+    remote_command_executor.run_remote_command("rm -f /shared/check_proc.out")
+    result = slurm_commands.submit_script(str(test_datadir / "check_mpi_process.sh"))
```

Comment:
you could also inline this script and use submit_command (the advantage is that we have the command in the code without having to open a separate file)